### PR TITLE
Add back HIP support to async net

### PR DESCRIPTION
**Pull request by bddppq**
We lost HIP support in last refactoring https://github.com/pytorch/pytorch/commit/620ece266880a0d3415134590cb4ba5e06f3a8e3

### CODE REVIEW

**Comment by bddppq** [comment]
https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger-test/4347/
https://ci.pytorch.org/jenkins/job/pytorch-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger/4339/

File: caffe2/core/net_async_base.cc [hunk 1]
```diff
@@ -178,24 +179,25 @@ TaskThreadPoolBase* AsyncNetBase::pool(const DeviceOption& device_option) {
...
-        caffe2::to_string(device_option.device_type()));
+        caffe2::to_string(device_type));
   }
 }
 
 int AsyncNetBase::stream(int task_id) {
   const auto& device_option = event(task_id).GetDeviceOption();
   int stream_id = 0;
-  if (device_option.device_type() == PROTO_CUDA) {
+  if (device_option.device_type() == PROTO_CUDA ||
```
Comment (dzhulgakov) [inline caffe2/core/net_async_base.cc]: should we have IsGpuDeviceType?

**Review by dzhulgakov — APPROVED** [review]

File: caffe2/core/net_async_base.cc (hunk 1 above)
Comment (bddppq) [reply caffe2/core/net_async_base.cc]: added

**Comment by bddppq** [comment]
https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger-test/4390/
https://ci.pytorch.org/jenkins/job/pytorch-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger/4382/

**Pull request closed without merging**